mne/viz/evoked.py

```diff
@@ -2129,8 +2132,20 @@ def plot_compare_evokeds(evokeds, picks=None, colors=None,
     # build up evokeds into a dict, if it's not already
     if isinstance(evokeds, Evoked):
         evokeds = [evokeds]
+
     if isinstance(evokeds, (list, tuple)):
-        evokeds = {str(idx + 1): evk for idx, evk in enumerate(evokeds)}
+        evokeds_copy = evokeds.copy()
+        evokeds = dict()
+        for evk_idx, evk in enumerate(evokeds_copy, start=1):
+            try:
```

Comment:
Use hasattr and avoid the try please   Thx

Reply:
Done in 21b41c6